great_expectations/experimental/datasources/pandas_datasource.py

```diff
@@ -201,19 +234,97 @@ def add_csv_asset(
         data_path: PathStr,
         regex: Union[str, re.Pattern],
         order_by: Optional[BatchSortersDefinition] = None,
-    ) -> CSVAsset:
+        **kwargs,  # TODO: update signature to have specific keys & types
+    ) -> CSVAsset:  # type: ignore[valid-type]
         """Adds a csv asset to this pandas datasource
 
         Args:
             name: The name of the csv asset
             data_path: Path to directory with csv files
             regex: regex pattern that matches csv filenames that is used to label the batches
             order_by: one of "asc" (ascending) or "desc" (descending) -- the method by which to sort "Asset" parts.
+            kwargs: Extra keyword arguments should correspond to ``pandas.read_csv`` keyword args
         """
         asset = CSVAsset(
             name=name,
             path=data_path,  # type: ignore[arg-type]  # str will be coerced to Path
             regex=regex,  # type: ignore[arg-type]  # str with will coerced to Pattern
             order_by=order_by or [],  # type: ignore[arg-type]  # coerce list[str]
+            **kwargs,
+        )
+        return self.add_asset(asset)
+
+    def add_json_asset(
```

Comment:
@Kilo59 I believe that the `FilesystemDataAsset` can support JSON as well -- seems no different than CSV, et al.

Reply:
The pandas method `read_csv` and `read_json` have different parameters. If we want to get rid of `kwargs`, display them explicitly in autocomplete (juptyer notebook, for example), and have error messages around them they will need to be different functions on the datasource.
It is true they could all create a `FilesystemDataAsset` in terms of reading them, but @Kilo59 has put validation of these different arguments in the different `DataAsset` subclasses, eg `CSVAsset` validates the input parameters that are CSV specific while `JSONAsset` validates all the arguments that are JSON specific.

Reply:
I think some of this might be clearer once I put the json schemas under source control.